packages/provider/wrapper.js

```diff
@@ -27,7 +27,7 @@ module.exports = {
 
     /* overwrite method */
     provider.send = this.send(originalSend, preHook, postHook);
-
+    provider.sendAsync = this.send(originalSend, preHook, postHook);
```

Comment:
Using `provider.sendAsync.bind(provider)` instead of `originalSend` seems like it'd be more correct.

Reply:
yes you're right. let me make the change.

Reply:
@davidmurdoch can you explain your reasoning? I don't understand why this is the case, should we be doing that with line 29 as well where we patch the `send` method?

Reply:
Yes, maybe I do not understand it correctly. Line 26 is where `originalSend` was assigned `this.send(originalSend, preHook, postHook)`.

So I thought for reliability it would be better to do `this.send(originalSendAsync, preHook, postHook)'. If I do the above `provider.sendAsync.bind(provider)`, would that just patch the provider's sendAsync method with the provider's send async method?

Reply:
@eggplantzzz  we are doing that with line 29 as well.